connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -152,12 +152,23 @@ message ExecutePlanRequest {
 message ExecutePlanResponse {
   string client_id = 1;
 
-  ArrowBatch arrow_batch = 2;
+  oneof response_type {
+    ArrowBatch arrow_batch = 2;
+
+    SqlCommandResult sql_command_result = 5;
+
+    // Support arbitrary result objects.
+    google.protobuf.Any extension = 999;
+  }
 
   // Metrics for the query execution. Typically, this field is only present in the last
   // batch of results and then represent the overall state of the query execution.
   Metrics metrics = 4;
 
+  message SqlCommandResult {
```

Comment:
How about we return an actual relation here? In that case the client can just plug this into their dataframe.
- Command would be a `LocalRelation` with any result contained inside of it.
- Other would be SQL.

Reply:
Done.